packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -199,72 +351,160 @@ export const makeEditGenerator = (
 		],
 	]);
 
-	return async (state: FuzzTestState): Promise<Operation | typeof done> => {
+	const buildOperation = (contents: FuzzTransactionEdit, treeIndex: number) => {
+		const operation: TransactionBoundary = {
+			type: "transaction",
+			contents,
+			treeIndex,
+		};
+		return operation;
+	};
+
+	return createAsyncGenerator<FuzzTransactionEdit, TransactionBoundary>(
+		baseTransactionEditGenerator,
+		buildOperation,
+	);
+};
+
+function createAsyncGenerator<Op, OpOut>(
+	baseGenerator: (state: FuzzTestState & TreeContext) => Promise<Op | typeof done>,
+	buildOperation: (contents: Op, treeIndex: number) => OpOut,
+): AsyncGenerator<OpOut, FuzzTestState> {
+	return async (state: FuzzTestState): Promise<OpOut | typeof done> => {
 		const trees = state.testTreeProvider.trees;
 		// does not include last tree, as we want a passive client
 		const treeIndex = trees.length === 1 ? 0 : state.random.integer(0, trees.length - 2);
 
-		const contents = await baseEditGenerator({
+		const contents = await baseGenerator({
 			...state,
 			treeIndex,
 		});
 		state.numberOfEdits += 1;
 		if (contents === done) {
 			return done;
 		}
-		return { type: "edit", contents, index: treeIndex };
+		return buildOperation(contents, treeIndex);
 	};
-};
+}
 
 export function makeOpGenerator(
-	editOpWeights: EditGeneratorOpWeights = defaultEditGeneratorOpWeights,
+	opWeights: EditGeneratorOpWeights = defaultEditGeneratorOpWeights,
 ): AsyncGenerator<Operation, FuzzTestState> {
-	const opWeights: AsyncWeights<Operation, FuzzTestState> = [
-		[makeEditGenerator(editOpWeights), 3],
-		[{ type: "synchronize" }, 1],
+	const generatorWeights: AsyncWeights<Operation, FuzzTestState> = [
+		[
+			makeEditGenerator(opWeights),
+			sumWeights([opWeights.delete, opWeights.insert, opWeights.setPayload]),
+		],
+		[{ type: "synchronize" }, opWeights.synchronize ?? 0],
+		[
+			makeTransactionEditGenerator(opWeights),
+			sumWeights([opWeights.abort, opWeights.commit, opWeights.start]),
+		],
 	];
-	return createWeightedAsyncGenerator(opWeights);
+	return createWeightedAsyncGenerator(generatorWeights);
+}
+
+function sumWeights(values: (number | undefined)[]): number {
+	let sum = 0;
+	for (const value of values) {
+		if (value !== undefined) {
+			sum += value;
+		}
+	}
+	return sum;
 }
 
 const moves = {
 	field: ["enterNode", "nextField"],
 	nodes: ["stop", "firstField"],
 };
 
-const nodePlaceType = ["beforeNode", "afterNode", "belowNode"];
+const sequenceFieldKey: FieldKey = brand("sequenceField");
+
+export interface FieldPathWithCount {
+	fieldPath: FieldUpPath;
+	fieldKey: FieldKey;
+	count: number;
+}
 
-function getRandomPlace(tree: ISharedTree, random: IRandom): UpPath {
-	const testerKey: FieldKey = brand("Test");
+function getExistingFieldPath(tree: ISharedTree, random: IRandom): FieldPathWithCount {
```

Comment:
it would be great to document some details on the strategy used for selecting a path here and its statistical properties.

Reply:
added doc comment